internal/bminventory/inventory.go

```
@@ -81,21 +91,84 @@ type debugCmd struct {
 	stepID string
 }
 
+type Command struct {
+	Name                string
+	Command             string
+	Arguments           []string
+	ExpectedArgumentNum int
+}
+
+type AgentCommands struct {
+	Commands []Command
+}
+
 type bareMetalInventory struct {
 	Config
 	imageBuildCmd []string
 	db            *gorm.DB
 	kube          client.Client
 	debugCmdMap   map[strfmt.UUID]debugCmd
 	debugCmdMux   sync.Mutex
+	agentCommands map[string]*Command
 }
 
-func NewBareMetalInventory(db *gorm.DB, kclient client.Client, cfg Config) *bareMetalInventory {
+func (b *bareMetalInventory) readAgentCommands() error {
+	contents, err := ioutil.ReadFile(AgentsCommandsFile)
+	if err != nil {
+		logrus.WithError(err).Errorf("Could not read file %s", AgentsCommandsFile)
+		return err
+	}
+	var agentCommands AgentCommands
+	err = yaml.Unmarshal(contents, &agentCommands)
+	if err != nil {
+		logrus.WithError(err).Errorf("Error unmarshaling %s", contents)
+		return err
+	}
+	b.agentCommands = make(map[string]*Command)
+	for _, cmd := range agentCommands.Commands {
+		b.agentCommands[cmd.Name] = &cmd
+	}
+	return nil
+}
+
+func (b *bareMetalInventory) getCommandByName(name string) *Command {
+	cmd, ok := b.agentCommands[name]
+	if !ok {
+		logrus.Errorf("Command %s was not found", name)
+		return nil
```

why not return an error?

---

The error was already logged.  If nil is returned then the step will mot be executed.  What else should be done?

---

i'm not talking about logging, we wanted to add a command and failed, something is wrong.